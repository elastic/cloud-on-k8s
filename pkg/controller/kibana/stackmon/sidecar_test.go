// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
// or more contributor license agreements. Licensed under the Elastic License 2.0;
// you may not use this file except in compliance with the Elastic License 2.0.

package stackmon

import (
	"context"
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
	kbv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/kibana/v1"
	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/defaults"
	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/stackmon/monitoring"
	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
)

func TestWithMonitoring(t *testing.T) {
	esRef := commonv1.ObjectSelector{Name: "sample", Namespace: "aerospace"}
	sampleKb := kbv1.Kibana{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "sample",
			Namespace: "aerospace",
		},
		Spec: kbv1.KibanaSpec{
			Version:          "7.14.0",
			ElasticsearchRef: esRef,
		},
	}
	monitoringEsRef := []commonv1.ObjectSelector{{Name: "monitoring", Namespace: "observability"}}
	logsEsRef := []commonv1.ObjectSelector{{Name: "logs", Namespace: "observability"}}

	fakeElasticUserSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-es-internal-users", Namespace: "aerospace"},
		Data:       map[string][]byte{"elastic-internal-monitoring": []byte("1234567890")},
	}
	fakeMetricsBeatUserSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-observability-monitoring-beat-es-mon-user", Namespace: "aerospace"},
		Data:       map[string][]byte{"aerospace-sample-observability-monitoring-beat-es-mon-user": []byte("1234567890")},
	}
	fakeLogsBeatUserSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-observability-logs-beat-es-mon-user", Namespace: "aerospace"},
		Data:       map[string][]byte{"aerospace-sample-observability-logs-beat-es-mon-user": []byte("1234567890")},
	}
	fakeEsHTTPCertSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-es-http-certs-public", Namespace: "aerospace"},
		Data: map[string][]byte{
			"tls.crt": []byte("7H1515N074r341C3r71F1C473"),
			"ca.crt":  []byte("7H1515N074r341C3r71F1C473"),
		},
	}
	fakeKbHTTPCertSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-kb-http-certs-public", Namespace: "aerospace"},
		Data: map[string][]byte{
			"tls.crt": []byte("7H1515N074r341C3r71F1C473"),
			"ca.crt":  []byte("7H1515N074r341C3r71F1C473"),
		},
	}
	fakeClient := k8s.NewFakeClient(&fakeElasticUserSecret, &fakeMetricsBeatUserSecret, &fakeLogsBeatUserSecret, &fakeEsHTTPCertSecret, &fakeKbHTTPCertSecret)

	monitoringAssocConf := commonv1.AssociationConf{
		AuthSecretName: "sample-observability-monitoring-beat-es-mon-user",
		AuthSecretKey:  "aerospace-sample-observability-monitoring-beat-es-mon-user",
		CACertProvided: true,
		CASecretName:   "sample-es-monitoring-observability-monitoring-ca",
		URL:            "https://monitoring-es-http.observability.svc:9200",
		Version:        "7.14.0",
	}
	logsAssocConf := commonv1.AssociationConf{
		AuthSecretName: "sample-observability-logs-beat-es-mon-user",
		AuthSecretKey:  "aerospace-sample-observability-logs-beat-es-mon-user",
		CACertProvided: true,
		CASecretName:   "sample-es-logs-observability-monitoring-ca",
		URL:            "https://logs-es-http.observability.svc:9200",
		Version:        "7.14.0",
	}

	tests := []struct {
		name                   string
		kb                     func() kbv1.Kibana
		containersLength       int
		podVolumesLength       int
		beatVolumeMountsLength int
	}{
		{
			name: "without monitoring",
			kb: func() kbv1.Kibana {
				return sampleKb
			},
			containersLength: 1,
		},
		{
			name: "with metrics monitoring",
			kb: func() kbv1.Kibana {
				sampleKb.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleKb)[0].SetAssociationConf(&monitoringAssocConf)
				return sampleKb
			},
			containersLength:       2,
			podVolumesLength:       4,
			beatVolumeMountsLength: 4,
		},
		{
			name: "with logs monitoring",
			kb: func() kbv1.Kibana {
				sampleKb.Spec.Monitoring.Metrics.ElasticsearchRefs = nil
				sampleKb.Spec.Monitoring.Logs.ElasticsearchRefs = monitoringEsRef
				monitoring.GetLogsAssociation(&sampleKb)[0].SetAssociationConf(&monitoringAssocConf)
				return sampleKb
			},
			containersLength:       2,
			podVolumesLength:       4,
			beatVolumeMountsLength: 4,
		},
		{
			name: "with metrics and logs monitoring",
			kb: func() kbv1.Kibana {
				sampleKb.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleKb)[0].SetAssociationConf(&monitoringAssocConf)
				sampleKb.Spec.Monitoring.Logs.ElasticsearchRefs = monitoringEsRef
				monitoring.GetLogsAssociation(&sampleKb)[0].SetAssociationConf(&logsAssocConf)
				return sampleKb
			},
			containersLength:       3,
			podVolumesLength:       7,
			beatVolumeMountsLength: 4,
		},
		{
			name: "with metrics and logs monitoring with different es ref",
			kb: func() kbv1.Kibana {
				sampleKb.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleKb)[0].SetAssociationConf(&monitoringAssocConf)
				sampleKb.Spec.Monitoring.Logs.ElasticsearchRefs = logsEsRef
				monitoring.GetLogsAssociation(&sampleKb)[0].SetAssociationConf(&logsAssocConf)
				return sampleKb
			},
			containersLength:       3,
			podVolumesLength:       8,
			beatVolumeMountsLength: 4,
		},
	}

	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			kb := tc.kb()
			builder := defaults.NewPodTemplateBuilder(corev1.PodTemplateSpec{}, kbv1.KibanaContainerName)
			_, err := WithMonitoring(context.Background(), fakeClient, builder, kb)
			assert.NoError(t, err)

			assert.Equal(t, tc.containersLength, len(builder.PodTemplate.Spec.Containers))
			for _, v := range builder.PodTemplate.Spec.Volumes {
				fmt.Println(v)
			}
			assert.Equal(t, tc.podVolumesLength, len(builder.PodTemplate.Spec.Volumes))

			if monitoring.IsMetricsDefined(&kb) {
				for _, c := range builder.PodTemplate.Spec.Containers {
					if c.Name == "metricbeat" {
						assert.Equal(t, tc.beatVolumeMountsLength, len(c.VolumeMounts))
					}
				}
			}
			if monitoring.IsLogsDefined(&kb) {
				for _, c := range builder.PodTemplate.Spec.Containers {
					if c.Name == "filebeat" {
						assert.Equal(t, tc.beatVolumeMountsLength, len(c.VolumeMounts))
					}
				}
			}
		})
	}
}
