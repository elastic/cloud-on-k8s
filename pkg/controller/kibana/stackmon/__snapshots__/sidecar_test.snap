
[TestWithMonitoring/without_monitoring - 1]
{
 "metadata": {},
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {}
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "kibana.k8s.elastic.co/monitoring-config-hash": "1132877422"
  }
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:7.14.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring/aerospace/sample/certs",
      "name": "kb-monitoring-local-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     },
     {
      "mountPath": "/usr/share/metricbeat/logs",
      "name": "metricbeat-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "name": "kb-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-local-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-http-certs-public"
    }
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-logs"
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_logs_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "kibana.k8s.elastic.co/monitoring-config-hash": "95064241"
  }
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:7.14.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/filebeat/logs",
      "name": "filebeat-logs"
     },
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "emptyDir": {},
    "name": "filebeat-logs"
   },
   {
    "name": "kb-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "emptyDir": {},
    "name": "kibana-logs"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-filebeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_and_logs_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "kibana.k8s.elastic.co/monitoring-config-hash": "1633666673"
  }
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:7.14.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring/aerospace/sample/certs",
      "name": "kb-monitoring-local-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     },
     {
      "mountPath": "/usr/share/metricbeat/logs",
      "name": "metricbeat-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:7.14.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/filebeat/logs",
      "name": "filebeat-logs"
     },
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "emptyDir": {},
    "name": "filebeat-logs"
   },
   {
    "name": "kb-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-logs-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-local-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-http-certs-public"
    }
   },
   {
    "emptyDir": {},
    "name": "kibana-logs"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-logs"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-filebeat-config"
    }
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_and_logs_monitoring_with_different_es_ref - 1]
{
 "metadata": {
  "annotations": {
   "kibana.k8s.elastic.co/monitoring-config-hash": "1902028541"
  }
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:7.14.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring/aerospace/sample/certs",
      "name": "kb-monitoring-local-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     },
     {
      "mountPath": "/usr/share/metricbeat/logs",
      "name": "metricbeat-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:7.14.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/logs/certs",
      "name": "kb-monitoring-584a4d-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/filebeat/logs",
      "name": "filebeat-logs"
     },
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "emptyDir": {},
    "name": "filebeat-logs"
   },
   {
    "name": "kb-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-584a4d-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-logs-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-local-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-http-certs-public"
    }
   },
   {
    "emptyDir": {},
    "name": "kibana-logs"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-logs"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-filebeat-config"
    }
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestMetricbeatConfig/no_monitoring - 1]

---

[TestMetricbeatConfig/with_metrics_monitoring - 1]
metricbeat:
    modules:
        - hosts:
            - https://localhost:5601
          metricsets:
            - stats
            - status
          module: kibana
          password: 1234567890
          period: 10s
          ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring/aerospace/sample/certs/ca.crt
            enabled: true
            verification_mode: certificate
          username: elastic-internal-monitoring
          xpack:
            enabled: true
output:
    elasticsearch:
        hosts:
            - https://monitoring-es-http.observability.svc:9200
        password: "1234567890"
        ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs/ca.crt
            verification_mode: certificate
        username: aerospace-sample-observability-monitoring-beat-es-mon-user
processors:
    - add_cloud_metadata: null
    - add_host_metadata: null

---

[TestMetricbeatConfig/with_monitoring_no_CA - 1]
metricbeat:
    modules:
        - hosts:
            - https://localhost:5601
          metricsets:
            - stats
            - status
          module: kibana
          password: 1234567890
          period: 10s
          ssl:
            enabled: true
            verification_mode: certificate
          username: elastic-internal-monitoring
          xpack:
            enabled: true
output:
    elasticsearch:
        hosts:
            - https://monitoring-es-http.observability.svc:9200
        password: "1234567890"
        ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs/ca.crt
            verification_mode: certificate
        username: aerospace-sample-observability-monitoring-beat-es-mon-user
processors:
    - add_cloud_metadata: null
    - add_host_metadata: null

---

[TestMetricbeatConfig/with_metrics_monitoring_no_TLS - 1]
metricbeat:
    modules:
        - hosts:
            - http://localhost:5601
          metricsets:
            - stats
            - status
          module: kibana
          password: 1234567890
          period: 10s
          ssl:
            enabled: false
            verification_mode: certificate
          username: elastic-internal-monitoring
          xpack:
            enabled: true
output:
    elasticsearch:
        hosts:
            - https://monitoring-es-http.observability.svc:9200
        password: "1234567890"
        ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs/ca.crt
            verification_mode: certificate
        username: aerospace-sample-observability-monitoring-beat-es-mon-user
processors:
    - add_cloud_metadata: null
    - add_host_metadata: null

---

[TestMetricbeatConfig/with_metrics_monitoring_and_basePath - 1]
metricbeat:
    modules:
        - basepath: /prefix
          hosts:
            - https://localhost:5601
          metricsets:
            - stats
            - status
          module: kibana
          password: 1234567890
          period: 10s
          ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring/aerospace/sample/certs/ca.crt
            enabled: true
            verification_mode: certificate
          username: elastic-internal-monitoring
          xpack:
            enabled: true
output:
    elasticsearch:
        hosts:
            - https://monitoring-es-http.observability.svc:9200
        password: "1234567890"
        ssl:
            certificate_authorities:
                - /mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs/ca.crt
            verification_mode: certificate
        username: aerospace-sample-observability-monitoring-beat-es-mon-user
processors:
    - add_cloud_metadata: null
    - add_host_metadata: null

---

[TestWithMonitoring/with_metrics_and_logs_monitoring_and_read_only_root_filesystem - 1]
{
 "metadata": {
  "annotations": {
   "kibana.k8s.elastic.co/monitoring-config-hash": "1902028541"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "kibana",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:7.14.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/monitoring/certs",
      "name": "kb-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring/aerospace/sample/certs",
      "name": "kb-monitoring-local-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     },
     {
      "mountPath": "/usr/share/metricbeat/logs",
      "name": "metricbeat-logs"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:7.14.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/kb-monitoring-association/observability/logs/certs",
      "name": "kb-monitoring-584a4d-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/filebeat/logs",
      "name": "filebeat-logs"
     },
     {
      "mountPath": "/usr/share/kibana/logs",
      "name": "kibana-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "emptyDir": {},
    "name": "filebeat-logs"
   },
   {
    "name": "kb-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-584a4d-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-logs-observability-monitoring-ca"
    }
   },
   {
    "name": "kb-monitoring-local-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-http-certs-public"
    }
   },
   {
    "emptyDir": {},
    "name": "kibana-logs"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "emptyDir": {},
    "name": "metricbeat-logs"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-filebeat-config"
    }
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-kb-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---
